website/static/js/project-settings-treebeard.js

```diff
@@ -0,0 +1,244 @@
+        var parent = tb.treeData.children[i];
+        tb.updateFolder(null, parent);
+        expandChildren(tb, parent.children);
+    }
+}
+
+function expandChildren(tb, children) {
+    var openParent = false;
+    for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        var parent = children[i].parent();
+        if (child.data.kind === 'event' && child.data.notificationType !== "adopt_parent") {
+            openParent = true;
+        }
+        if (child.children.length > 0) {
+            expandChildren(tb, child.children);
+        }
+    }
+    if (openParent) {
+        openAncestors(tb, children[0]);
+    }
+}
+
+function openAncestors (tb, item) {
+    var parent = item.parent();
+    if(parent && parent.id > 0) {
+        tb.updateFolder(null, parent);
+        openAncestors(tb, parent);
+    }
+}
+
+function subscribe(id, event, notification_type) {
+    var payload = {
+                'id': id,
+                'event': event,
+                'notification_type': notification_type
+    };
+    $osf.postJSON(
+        '/api/v1/settings/subscribe/',
+        payload
+    ).fail(function() {
+        bootbox.alert('Could not update notification preferences.')
+    });
+}
+
+
+function displayParentNotificationType(item){
+    var notificationTypeDescriptions = {
+        'email_transactional': 'Emails',
+        'email_digest': 'Email Digest',
+        'adopt_parent': 'Adopt setting from parent project',
+        'none': 'None'
+    };
+
+    if (item.data.parent_notification_type) {
+        if (item.parent().parent().parent() === undefined) {
+            return "(" + notificationTypeDescriptions[item.data.parent_notification_type] + ")";
+        }
+    }
+    return "";
+}
+
+
+function ProjectNotifications(data) {
+
+    //  Treebeard version
+    var tbOptions = {
+        divID: 'grid',
+        filesData: data,
+        rowHeight : 40,         // user can override or get from .tb-row height
+        resolveToggle: resolveToggle,
+        paginate : false,       // Whether the applet starts with pagination or not.
+        paginateToggle : false, // Show the buttons that allow users to switch between scroll and paginate.
+        uploads : false,         // Turns dropzone on/off.
+        resolveIcon : resolveIcon,
+        onload: function () {
+            var tb = this;
+            expandOnLoad.call(tb);
+        },
+        columnTitles : function notificationColumnTitles(item, col) {
+            return [
+                {
+                    title: "Project",
+                    width: "60%",
+                    sortType : "text",
+                    sort : false
+                },
+                {
+                    title: "Notification Type",
+                    width : "40%",
+                    sort : false
+
+                }
+            ]},
+        resolveRows : function notificationResolveRows(item){
+            var default_columns = [];
```

Comment:
Use `camelCase` for variable names. Also, `default_columns` isn't really an appropriate name for this variable (it's not a "default"). It is the return value of this function, which is an array of objects describing columns. So maybe `columns`, or `ret`?